### PR TITLE
(SIMP-1695) Refactored to work better with autofs

### PR DESCRIPTION
**Pull request by trevor-vaughan**
This updates the module to work better with autofs as well as the
upcoming simp_nfs profile module.

SIMP-1695 #comment Updated to work better with autofs and simp_nfs

### CODE REVIEW

**Comment by trevor-vaughan** [comment]
NOTE: The stunnel acceptance test need to be updated!

File: spec/acceptance/suites/stunnel/03_stunnel_test_spec.rb [hunk 1]
```diff
@@ -53,10 +55,11 @@
...
-simp_options::tcpwrappers : true
+simp_options::haveged : true
 simp_options::kerberos : false
 simp_options::pki : true
-simp_options::haveged : true
+simp_options::stunnel : true
+simp_options::tcpwrappers : true
 simp_options::trusted_nets : ['ALL']
 
 pki_dir : '/etc/pki/simp-testing/pki'
```
Comment (jeannegreulich) [inline spec/acceptance/suites/stunnel/03_stunnel_test_spec.rb]: You should just need to set app_pki_external_source to this.

**Comment by trevor-vaughan** [comment]
@jeannegreulich This should work now.

**Comment by trevor-vaughan** [comment]
@jeannegreulich Never mind...apparently have a typo even though the acceptance tests work. If you can fix that, it would be appreciated. If not, I'll get it tonight.

**Pull request merged**